### PR TITLE
Adding basic tests

### PR DESCRIPTION
**Pull request by skatkov**
- Adding basic tests.
- `#ensure_state_may_transition_to!` method now actually verifies that an attribute can transition to a new state.

### CODE REVIEW

File: state_machine_enum.gemspec [hunk 1]
```diff
@@ -33,5 +33,7 @@ Gem::Specification.new do |spec|
   spec.executables = spec.files.grep(%r{\Aexe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "activesupport", ">= 6.0"
+  spec.add_dependency "activesupport", "~> 7"
+  spec.add_development_dependency "sqlite3", "~> 1.4"
+  spec.add_development_dependency "rails", "~> 7"
```
Comment (julik) [inline state_machine_enum.gemspec:38]: Let's not do this - depending just on activerecord is sufficient. The Gemfile.lock becomes huge and we don't need the entirety of Rails at all

File: test/test_state_machine_enum.rb [hunk 2]
```diff
@@ -2,12 +2,71 @@
...
+
+  def test_ensure_state_may_transition_to!
+    u = User.create!(state: "active")
+
+    assert_nil u.ensure_state_may_transition_to!(:banned)
+
+    assert_raises(StateMachineEnum::InvalidState) { u.ensure_state_may_transition_to!("active") }
+    assert_raises(StateMachineEnum::InvalidState) { u.ensure_state_may_transition_to!(:active) }
+
+    # TODO: This method will not actually check if state can transition to the target state, it only verifies that this is not a current state.
```
Comment (julik) [inline test/test_state_machine_enum.rb:45]: We can change that behavior I think. Should we rename this to `ensure_state_may_transition_to_or_is!`? and then do the actual check that it is permitted to transition?

**Review by julik — CHANGES_REQUESTED** [review]
Great start!

File: test/test_state_machine_enum.rb (hunk 2 above)
Comment (skatkov) [reply test/test_state_machine_enum.rb]: Name is fine, but the fact that we don't really check that state can transition was an issue. I have fixed that.

**Review by julik — APPROVED** [review]

**Pull request merged**